VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsActivity.java

```diff
@@ -74,6 +74,11 @@ public Fragment getFirstFragment() {
         } else {
             fragment = courseDiscussionPostsThreadFragment;
         }
+        if (fragment.getArguments() == null) {
+            Bundle args = new Bundle();
+            args.putSerializable(Router.EXTRA_ENROLLMENT_COURSE_DATA, courseData);
+            fragment.setArguments(args);
```

Comment:
OK, how about adding a TODO note here explaining the necessity of setting the arguments here, and noting that we should eventually move this logic to the base class?